### PR TITLE
#164798294 Users should be able to rate articles

### PR DESCRIPTION
**Pull request by anyatibrian**
#### What does this PR do?
Ensures users are able to rate articles 

#### Description of Task to be completed?
- add endpoints for rating articles
- Write tests for 
- validate the rate endpoint

#### How should this be manually tested?
- when you hit `localhost:8000/api/v1/articles/bentl-eaasy-0x5/rate/`
with this data.
   `{
	"ratings":4
	}`

 - and expect
`{
    "message": "thanks for rating this article"
}`
- with invalide data ie
  `{
	"ratings":7
  }`
 expect this 
`{
    "errors": {
        "ratings": [
            "Ratings should be numbers between 0-5"
        ]
    }
}`
- when you hit `localhost:8000/api/v1/articles/`

```
{
        "id": 7,
        "author": {
            "username": "briananyati",
            "bio": "",
            "email": "brian.ogwal@andela.com",
            "full_name": "",
            "image": "",
            "followers_no": 0,
            "following_no": 0
        },
        "average_rating": 3.7,
        "title": "Bentl eaasy",
        "description": "JuBentlsey",
        "body": "I like ntsley",
        "createdAt": "2019-04-11T03:30:18.940653Z",
        "updatedAt": "2019-04-11T03:30:18.940681Z",
        "slug": "bentl-eaasy-0x6"
    }

```
#### Relevant Pivotal Tracker story
[#164798294](https://www.pivotaltracker.com/story/show/164798294)
#### Screen shorts
when your fetch all the articles
Screenshot 2019-04-11 at 13 04 27
when rating an article
Screenshot 2019-04-11 at 13 16 42

### CODE REVIEW

File: authors/apps/articles/models.py [hunk 1]
```diff
@@ -15,22 +16,38 @@ class Articles(models.Model):
...
         return self.title
 
 
+class RateArticles(models.Model):
+    """
+    model for rating  article
+    """
+    user = models.ForeignKey(Profile, on_delete=models.CASCADE)
+    article = models.ForeignKey(Articles, on_delete=models.CASCADE)
+    ratings = models.DecimalField(max_length=20, max_digits=5, decimal_places=2)
```
Comment (e-ian) [inline authors/apps/articles/models.py]: @anyatibrian I think this was recommended to take integer values. Please update

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -94,16 +93,41 @@ def delete(self, request, *args, **kwargs):
...
+    view for rating articles
+    """
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = serializers.RateArticleSerializer
+
+    def post(self, request, **kwargs):
+        slug = kwargs.pop('slug')
+        user = request.user
+        article = Articles.objects.filter(slug=slug).first()
+        data = {'user': article.author.id, 'article': article.pk, 'ratings': request.data.get('ratings')}
```
Comment (e-ian) [inline authors/apps/articles/views.py]: These lines are too long @anyatibrian. Consider linting

**Review by e-ian — COMMENTED** [review]
@anyatibrian do linting. Some lines are too long. Readabilty counts.

File: authors/apps/articles/views.py (hunk 2 above)
Comment (anyatibrian) [reply authors/apps/articles/views.py]: @e-ian resolved

**Review by e-ian — APPROVED** [review]
LGTM

**Review by inno-asiimwe — APPROVED** [review]
LGTM

**Pull request merged**